src/blueprint_provider/default_vm_blueprint_provider.cpp

```diff
@@ -41,39 +41,117 @@ namespace mpl = multipass::logging;
 namespace
 {
 const QString github_blueprints_archive_name{"multipass-blueprints.zip"};
-const QString blueprint_dir_version{"v1"};
+
+// The folders where to read definitions. The list is sorted by precedence.
+constexpr auto version_v1{"v1"};
+constexpr auto version_v2{"v2"};
+const QStringList blueprint_dir_versions{version_v2, version_v1};
+
 constexpr auto category = "blueprint provider";
 
-auto blueprints_map_for(const std::string& archive_file_path, bool& needs_update)
+static constexpr auto bad_conversion_template{"Cannot convert \'{}\' key for the {} Blueprint"};
+const auto runs_on_key{"runs-on"};
+const auto instances_key{"instances"};
+
+bool runs_on(const std::string& blueprint_name, const YAML::Node& blueprint_node, const std::string& arch)
+{
+    if (blueprint_node["blueprint-version"].as<std::string>() == version_v1)
+    {
+        if (blueprint_node[runs_on_key])
+        {
+            try
+            {
+                std::vector<std::string> runs_on = blueprint_node[runs_on_key].as<std::vector<std::string>>();
```

Comment:
Hey @townsend2010, sorry, I can't get what you mean here. Are you referring to `runs_on_key`? It's compiling and running fine on my computer.

Reply:
I mean does `const auto runs_on =` not work instead of explicitly using the type?  I prefer the `auto` convention, but if it doesn't work, then I understand.

Reply:
Oh, right, sorry. Yes, `const auto` works fine :). I corrected it.